### PR TITLE
Add static private class field support

### PR DESCRIPTION
**Pull request by jridgewell**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| Breaking change?  | no
| New feature?      | yes
| Deprecations?     | no
| Spec compliancy?  | yes
| Tests added/pass? | yes
| Fixed tickets     | 
| License           | MIT

Private class fields can be static (according to the [unified proposal](https://github.com/tc39/proposal-class-fields)).

### CODE REVIEW

**Comment by xtuc** [comment]
Nice job @jridgewell.

Do you think we could merge classProperties and classPrivateProperties plugins in Babylon? Since it just adds the static syntax.

**Comment by jridgewell** [comment]
I'll do that in a second PR.

**Comment by diervo** [comment]
@xtuc @jridgewell 
We should wait for merging both plugins until publicFields get to Stage3, which hopefully will happen in the next TC39 meeting (in about a month). 

There is still some debate on the committee on wether `this.#x` will be required and so let's wait for the final version.

Decorators might be another moving piece that is very intertwined, but I guess is mostly implementation details.

Lastly, just an FYI I think @littledan wants to present private methods, so let's see which Stage that lands... :)

File: test/fixtures/experimental/class-private-properties/static/actual.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class A {
+  static #x;
+  static #y = 1;
+}
```
Comment (littledan) [inline test/fixtures/experimental/class-private-properties/static/actual.js]: Have comma-separated field declarations landed yet? If so, it'd be nice to have tests that demonstrate that you can mix static public and private field declarations in a comma-separated list.

**Comment by littledan** [comment]
This looks great. I don't see why it should wait until Stage 3 to land--I presented static private fields to the committee in the May meeting with this syntax and didn't hear any concerns. I think @hzoo is doing a lot of work to communicate to users that Stage 2 proposals are unstable.

For decorators, we unfortunately don't have spec text yet which combines it with fields; hope to fix that soon. Private methods haven't been presented yet, so I guess they will wait a bit longer to add to Babel.

File: test/fixtures/experimental/class-private-properties/static/actual.js (hunk 1 above)
Comment (diervo) [inline test/fixtures/experimental/class-private-properties/static/actual.js]: PR is pending

**Comment by diervo** [comment]
@littledan I was not talking about this PR, this is good to go, the refactor to combine public/private fields is what Im saying to postpone til we figure out the `this.#x` syntax.

**Pull request merged**